client/src/main/java/org/eclipse/hono/client/impl/HonoClientImpl.java

```diff
@@ -592,7 +598,17 @@ private void reconnect(
         return result;
     }
 
-    private Future<RequestResponseClient> newCredentialsClient(final String tenantId) {
+    /**
+     * Custom implementation of {@link CredentialsClient} can be instantiated by overriding this
```

Comment:
Please re-phrase this JavaDoc.
The first sentence should describe shortly what this method's purpose is. Subsequent paragraphs can then go into specifics and then you may add recommendations for properly overriding the method at the end.
See http://www.oracle.com/technetwork/articles/java/index-137868.html